### PR TITLE
RESTful API Search integration

### PR DESCRIPTION
**Pull request by pgfrank1**
I have made the search API have the ability to receive a user searchTerm and apply that term to multiple variables within the DataItems.

### CODE REVIEW

**Review by pgfrank1 — COMMENTED** [review]
FYI, a lot of the changes are documentation updates

File: pom.xml [hunk 1]
```diff
@@ -1,193 +1,193 @@
...
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-api</artifactId>
+      <version>${junit-jupiter-api-version}</version>
+    </dependency>
+  </dependencies>
+  <build>
+    <finalName>untitled</finalName>
+    <pluginManagement><!-- lock down plugins versions to avoid using Maven defaults (may be moved to parent pom) -->
+      <plugins>
```
Comment (pgfrank1) [inline pom.xml:160]: The biggest change here was making sure that the plugins are a specified version. These versions are also in Paula's example project. This also removed the Surefire dependencies as it caused the Unit Tests to run upon each rebuild.

**Review by erickr19 — COMMENTED** [review]
These changes look good. @kuexiong when you're finished reviewing them could you please approve and merge them?

**Review by kuexiong — APPROVED** [review]

**Review by pgfrank1 — COMMENTED** [review]
Reviewed for approval

**Review by erickr19 — APPROVED** [review]

**Pull request merged**